src/basic/time-util.c

```diff
@@ -1305,11 +1309,30 @@ bool timezone_is_valid(const char *name) {
                 return false;
 
         t = strjoina("/usr/share/zoneinfo/", name);
-        if (stat(t, &st) < 0)
+
+        fd = open(t, O_RDONLY|O_CLOEXEC);
+        if (fd < 0) {
+                log_debug_errno(errno, "Failed to open timezone file '%s': %m", t);
+                return false;
+        }
+
+        r = fd_verify_regular(fd);
+        if (r < 0) {
+                log_debug_errno(r, "Timezone file '%s' is not  a regular file: %m", t);
                 return false;
+        }
+
+        r = loop_read_exact(fd, buf, 4, false);
+        if (r < 0) {
+                log_debug_errno(r, "Failed to read from timezone file '%s': %m", t);
+                return false;
+        }
 
-        if (!S_ISREG(st.st_mode))
+        /* Magic from tzfile(5) */
+        if (memcmp(buf, "TZif", 4) != 0) {
+                log_debug("Timezone file '%s' has wrong magic bytes", t);
```

Comment:
Making this log_debug means that it's hidden by default. Imagine that fistboot is running, and you get asked to enter a timezone, and it just says "Entered data invalid" over and over. I think this one and the checks above should be `log_warning`, because those are all rather unexpected conditions and we should log about them.

Reply:
`timezone_is_valid` is called from `sd_dhcp_server_set_timezone` and `dhcp_lease_parse_options`, and since those are library functions, we should keep logging at debug level from those. I guess this means that `timezone_is_valid` should take the log level as an argument.